### PR TITLE
nixos/stevenblack: init

### PR DESCRIPTION
**Pull request by moni-dz**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/config/stevenblack.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    ++ optionals porn [ "porn" ]
+    ++ optionals social [ "social" ];
+
+  hostsPath = "${pkgs.stevenblack-blocklist}/alternates/" + concatStringsSep "-" activatedHosts + "/hosts";
+in
+{
+  options.networking.stevenblack = {
+    enable = mkEnableOption "Enable the stevenblack hosts file blocklist.";
+
+    block = {
```
Comment (Artturin) [inline nixos/modules/config/stevenblack.nix:20]: Would be better as a list of enums iirc it's called

Comment (fortuneteller2k) [reply nixos/modules/config/stevenblack.nix:20]: Hmm, I don't think it's possible to do it that way, the blocklist allows for multiple combinations and they have to be in a specific order (i.e. if `fakenews` is activated then it has to be the first in the `activatedHosts` list so the folder exists [here](https://github.com/StevenBlack/hosts/tree/master/alternates)).

Comment (Artturin) [reply nixos/modules/config/stevenblack.nix]: i rebased the pr and switched to use elem

Comment (fortuneteller2k) [reply nixos/modules/config/stevenblack.nix]: looks good

**Pull request merged**